sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -57,6 +58,18 @@ internal set
             }
         }
 
+        internal static bool TryGetInternalInstance([NotNullWhen(true)] out IDeploymentInternal? instance)
+        {
+            if (_instance.Value != null)
+            {
+                instance = _instance.Value.Internal;
+                return true;
+            }
+
+            instance = null;
+            return false;
+        }
+
```

Comment:
Since this is only used by the one place currently, this could be replaced with a more specific method for registering a task. E.g. something like

```csharp
// If we care about reporting whether we are able to register the task or not
internal static bool TryRegisterTask(string description, Task task)
{
    if (_instance.Value is null)
    {
        return false;
    }

    _instance.Value.Internal.Runner.RegisterTask(description, task);
    return true;
}

// or simply
internal static void RegisterTask(string description, Task task)
{
    if (_instance.Value != null)
    {
        _instance.Value.Internal.Runner.RegisterTask(description, task);
    }
}
```